src/libsync/discovery.cpp

```diff
@@ -242,17 +242,15 @@ bool ProcessDirectoryJob::handleExcluded(const QString &path, const QString &loc
             if (item->_file.endsWith(QLatin1Char('.'))) {
                 item->_errorString = tr("File names ending with a period are not supported on this file system.");
             } else {
-                char invalid = '\0';
-                for (const char x : QByteArrayLiteral("\\:?*\"<>|")) {
-                    if (item->_file.contains(QLatin1Char(x))) {
-                        invalid = x;
+                static const QString unsupportedCharacters = QStringLiteral("\\:?*\"<>|");
```

Comment:
the string literal is already static https://code.woboq.org/qt5/qtbase/src/corelib/text/qstringliteral.h.html#_M/QStringLiteral